instrumentation/jdbc/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcSingletons.java

```diff
@@ -17,17 +19,20 @@
 import io.opentelemetry.instrumentation.jdbc.internal.JdbcNetAttributesGetter;
 import io.opentelemetry.javaagent.bootstrap.internal.CommonConfig;
 import io.opentelemetry.javaagent.bootstrap.internal.InstrumentationConfig;
+import javax.sql.DataSource;
 
 public final class JdbcSingletons {
   private static final String INSTRUMENTATION_NAME = "io.opentelemetry.jdbc";
 
-  private static final Instrumenter<DbRequest, Void> INSTRUMENTER;
+  private static final Instrumenter<DbRequest, Void> STATEMENT_INSTRUMENTER;
+  public static final Instrumenter<DataSource, Void> DATASOURCE_INSTRUMENTER =
+      createDataSourceInstrumenter(GlobalOpenTelemetry.get());
 
   static {
     JdbcAttributesGetter dbAttributesGetter = new JdbcAttributesGetter();
     JdbcNetAttributesGetter netAttributesGetter = new JdbcNetAttributesGetter();
 
-    INSTRUMENTER =
+    STATEMENT_INSTRUMENTER =
```

Comment:
should I use the createStatementInstrumenter here as well?

Reply:
Probably not, since the javaagent setup is a bit more complex.